src/julia-parser.scm

```diff
@@ -1543,16 +1547,30 @@
                                               '()
                                               (cdr catch-block))))
                             (if var? var '(false))
-                            finalb)))))
+                            finalb
+                            elseb)))))
              ((and (eq? nxt 'finally)
                    (not finalb))
-              (let ((fb (if (eq? (require-token s) 'catch)
+              (let ((fb (if (eq? (require-token s) '(catch else))
                             '(block)
                             (parse-block s))))
                 (loop (require-token s)
                       catchb
                       catchv
-                      fb)))
+                      fb
+                      elseb)))
+             ((and (eq? nxt 'else)
+                   (not elseb))
+              (if (or (not catchb) finalb)
+                  (error "else inside try block needs to be immediately after catch"))
+              (let ((eb (if (eq? (require-token s) '(catch finally))
```

Comment:
```suggestion
              (let ((eb (if (memq (require-token s) '(catch finally))
```

Reply:
though we don't need to look for `catch` here since it can't come after `else` anyway?

Reply:
That's true.
```suggestion
              (let ((eb (if (eq? (require-token s) 'finally)
```